src/test/java/soolutions/hometaskapp/account/TransferTests.java

```diff
@@ -0,0 +1,77 @@
+package soolutions.hometaskapp.account;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import soolutions.hometaskapp.account.Account;
+import soolutions.hometaskapp.account.BasicAccount;
+import soolutions.hometaskapp.account.Deposit;
+import soolutions.hometaskapp.account.Transaction;
+import soolutions.hometaskapp.account.Transfer;
+import soolutions.hometaskapp.account.Transfer.InvalidTransactionException;
+import soolutions.hometaskapp.common.Amount;
+import soolutions.hometaskapp.user.BasicUser;
+
+public class TransferTests {
+    private AccountFactory dollarAccountFactory;
+
+    @BeforeEach
+    void setup() {
+        dollarAccountFactory = new AccountFactory("USD");
+    }
+
+    @Test
+    void transferShouldFailForClosedAccounts() {
+        Transfer transfer = new Transfer();
```

Comment:
Do you think it would be better to pass the Accounts and the Amount into the constructor of a Transfer, instead of passing them during `apply`? This might represent the idea of a single transfer instance better.

Reply:
"This might represent the idea of a single transfer instance better" can you please elaborate it?

Reply:
When we have an instance of `Transfer` sitting around in memory, and he can process various transfer, then he behaves like a utility class or service object. He exists solely to help other object do something they can't do. 

```
Transfer transfer = new Transfer();
transfer.apply(... first transfer ...);
transfer.apply(... second transfer ...);
transfer.apply(... third transfer ...);
```

When we make a fresh instance every time we need to make a transfer, the data is encapsulated and he represents a specific transfer at a specific point in time. He is an object. 


```
new Transfer(... first transfer ...).execute();
new Transfer(... second transfer ...).execute();
new Transfer(... third transfer ...).execute();
```

Here the three instances of transfer represent three times that a transfer happened in the system. It's a one-to-one relationship.

Reply:
That seems a valid argument, but isn't `Transfer` is a `Service Object`? so, according to me you want to say that `Service Object` need not to be single instance.

Reply:
Yes, right now he is a Service Object.

I was just curious if you like that design best. Both designs will satisfy the current use case, but the will require different things of the caller. That's why I asked "Do you think it would be better...?" It's a consideration -- a discussion point. It's not a recommendation or complaint.

Reply:
I like `Transfer` as a one to one relation because it seems to fit the mental model but I don't know what its pros and cons be.